tests/MongoDB.Driver.Tests/MongoCollectionImplTests.cs

```diff
@@ -3297,6 +3299,35 @@ void assertException(Exception exception)
             e.ParamName.Should().Be("pipeline");
         }
 
+        [SkippableFact]
+        public void Watch_should_support_full_document_with_duplicate_elements()
```

Comment:
This is a full blown integration test.

Originally I thought this would require multiple threads but I then realized it didn't really, so the test is easier than I originally expected.